src/provider/provider_os_memory.c

```diff
@@ -794,6 +837,7 @@ static umf_result_t os_allocation_merge(void *provider, void *lowPtr,
 
 typedef struct os_ipc_data_t {
     int pid;
+    char shm_name[MAX_NAME];
```

Comment:
You are extending `os_ipc_data_t` by `MAX_NAME` bytes, but it is needed only if named shared memory is used. Perhaps we need two IPC data structures for anonymous and named shared memory.

Reply:
Done